### PR TITLE
Generate events when programatically selecting a tree item in wxQT

### PR DESCRIPTION
**Pull request by ffa-grahamdawes**
The previous implementation of wxTreeCtrl::SelectItem for wxQT update the selection but didn't generate any events.  This minor change corrects this.

### CODE REVIEW

**Comment by oneeyeman1** [comment]
@ffa-grahamdawes ,
What is the other platforms behavior?
The notion of wx is to not generate events when there is a programmatic change...

**Comment by ffa-grahamdawes** [comment]
wxMSW and wxGTK generate the events.  These can be seen from the treectrl sample program using "select root" menu item.    Unfortunately, I don't have access to MacOS here and I can't get wx to build on my Mac at home (I'm using a beta version of MacOS + XCode)

File: src/qt/treectrl.cpp [hunk 1]
```diff
@@ -244,6 +244,12 @@ class wxQTreeWidget : public wxQtEventSignalHandler<QTreeWidget, wxTreeCtrl>
         return m_placeHolderImage;
     }
 
+    void select(QTreeWidgetItem* item, QItemSelectionModel::SelectionFlag selectionFlag)
+    {
+        const QModelIndex index = indexFromItem(item);
```
Comment (catalinr) [inline src/qt/treectrl.cpp]:
```suggestion
        const QModelIndex &index = indexFromItem(item);
```

File: src/qt/treectrl.cpp [hunk 2]
```diff
@@ -1164,19 +1170,13 @@ void wxTreeCtrl::Toggle(const wxTreeItemId& item)
 void wxTreeCtrl::Unselect()
 {
     QTreeWidgetItem *current = m_qtTreeWidget->currentItem();
-
-    if ( current != NULL )
-        current->setSelected(false);
+    if (current != NULL)
```
Comment (catalinr) [inline src/qt/treectrl.cpp]:
```suggestion
    if ( current != NULL )
```
actually this line should not have been changed at all

File: src/qt/treectrl.cpp [hunk 3]
```diff
@@ -1189,7 +1189,15 @@ void wxTreeCtrl::SelectItem(const wxTreeItemId& item, bool select)
     }
 
     QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
-    qTreeItem->setSelected(select);
+
+    if (qTreeItem != NULL)
```
Comment (catalinr) [inline src/qt/treectrl.cpp]:
```suggestion
    if ( qTreeItem )
```

File: src/qt/treectrl.cpp [hunk 4]
```diff
@@ -1201,7 +1209,7 @@ void wxTreeCtrl::SelectChildren(const wxTreeItemId& parent)
 
     for ( int i = 0; i < childCount; ++i )
     {
-        qTreeItem->child(i)->setSelected(true);
+         m_qtTreeWidget->select(qTreeItem->child(i),  QItemSelectionModel::Select);
```
Comment (catalinr) [inline src/qt/treectrl.cpp]:
```suggestion
        m_qtTreeWidget->select(qTreeItem->child(i), QItemSelectionModel::Select);
```

**Review by catalinr — CHANGES_REQUESTED** [review]

**Comment by vadz** [comment]
We actually have this covered by the unit tests in `tests/controls/treectrltest.cpp`, it would be great if wxQt could pass them (but this requires a working `wxUIActionSimulator` and I don't know how well does it work in wxQt -- it seems to be implemented, at least).

Anyhow, thanks for fixing this, merged into #1552 too.

**Pull request closed without merging**